code/control/DeployDispatcher.php

```diff
@@ -80,66 +80,74 @@ public function history(SS_HTTPRequest $request) {
 			return $this->getAPIResponse(['list' => []], 200);
 		}
 
-		$currentBuild = $this->environment->CurrentBuild();
-
 		foreach ($list as $deployment) {
-			$data[] = [
-				'ID' => $deployment->ID,
-				'CreatedDate' => $deployment->Created,
-				'Branch' => $deployment->Branch,
-				'Tags' => $deployment->getTags()->toArray(),
-				'Changes' => $deployment->getDeploymentStrategy()->getChanges(),
-				'SHA' => $deployment->SHA,
-				'CommitMessage' => $deployment->getCommitMessage(),
-				'CommitURL' => $deployment->getCommitURL(),
-				'Deployer' => $deployment->Deployer()->getName(),
-				'Approver' => $deployment->Approver()->getName(),
-				'State' => $deployment->State,
-				'IsCurrentBuild' => $currentBuild ? ($deployment->ID === $currentBuild->ID) : null
-			];
+			$data[] = $this->getDeploymentData($deployment);
 		}
 
 		return $this->getAPIResponse([
 			'list' => $data,
-			'pagelength' => $list->getPageLength(),
-			'totalpages' => $list->TotalPages(),
-			'currentpage' => $list->CurrentPage()
+			'page_length' => $list->getPageLength(),
+			'total_pages' => $list->TotalPages(),
+			'current_page' => $list->CurrentPage()
 		], 200);
 	}
 
 	/**
-	 * @return SS_HTTPResponse
+	 * @param \SS_HTTPRequest $request
+	 * @return \SS_HTTPResponse
 	 */
 	public function currentbuild(SS_HTTPRequest $request) {
 		$currentBuild = $this->environment->CurrentBuild();
 		if (!$currentBuild) {
-			return $this->getAPIResponse(['build' => []], 200);
+			return $this->getAPIResponse(['deployment' => []], 200);
 		}
+		return $this->getAPIResponse(['deployment' => $this->getDeploymentData($currentBuild)], 200);
+	}
 
-		return $this->getAPIResponse(['build' => [
-			'ID' => $currentBuild->ID,
-			'CreatedDate' => $currentBuild->Created,
-			'Branch' => $currentBuild->Branch,
-			'Tags' => $currentBuild->getTags()->toArray(),
-			'SHA' => $currentBuild->SHA,
-			'CommitMessage' => $currentBuild->getCommitMessage(),
-			'CommitURL' => $currentBuild->getCommitURL(),
-		]], 200);
+	/**
+	 * @param \SS_HTTPRequest $request
+	 * @return \SS_HTTPResponse
+	 */
+	public function show(SS_HTTPRequest $request) {
+		$deployment = DNDeployment::get()->byId($request->param('ID'));
+		if (!$deployment || !$deployment->exists()) {
+			return $this->getAPIResponse(['message' => 'This deployment does not exist'], 404);
+		}
+		if (!$deployment->canView()) {
```

Comment:
I guess this will have to be rethought with approvers and other type of users later?